model/src/test/kotlin/com/testerum/model/resources/http/response/XmlBodyHttpResponseTest.kt

```diff
@@ -0,0 +1,188 @@
+
+        expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.bookstore.book[1].getText(1)""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("clar ca da", expressionResult)
+
+        expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.bookstore.book[1].getText(2)""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("merge si asa", expressionResult)
+    }
+
+    @Test
+    fun getChildMethodOnXmlNode() {
+        val xml = """
+            <bookstore>
+              <book>
+                <title lang="en">Harry Potter</title>
+                <price>29.99</price>
+              </book>
+
+              <book>
+                <title lang="nl">Learning XML</title>
+                <price>39.95</price>
+              </book>
+            </bookstore>
+        """.trimIndent()
+
+        val expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.bookstore.getChild("book")[0].getChild("title").getAttr("lang")""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+
+        Assertions.assertEquals("en", expressionResult)
+    }
+
+    @Test
+    fun evaluateXPathToGetAttribute() {
+        val xml = """
+            <bookstore>
+              <book>
+                <title lang="en">Harry Potter</title>
+                <price>29.99</price>
+              </book>
+
+              <book>
+                <title lang="nl">Learning XML</title>
+                <price>39.95</price>
+              </book>
+            </bookstore>
+        """.trimIndent()
+
+        var expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.xpath("/*/book[1]/title/@lang")""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("en", expressionResult)
+
+        expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.xpath("/*/book[2]/title/@lang")""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("nl", expressionResult)
+    }
+
+    @Test
+    fun evaluateXPathToGetText() {
+        val xml = """
+            <bookstore>
+              <book>
+                <title lang="en">Harry Potter</title>
+                <price>29.99</price>
+              </book>
+
+              <book>
+                <title lang="nl">Learning XML</title>
+                <price>39.95</price>
+              </book>
+            </bookstore>
+        """.trimIndent()
+
+        var expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.xpath("/*/book[1]/title/text()")""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("Harry Potter", expressionResult)
+
+        expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.xpath("/*/book[2]/title/text()")""",
```

Comment:
``xpath`` should be available not only on the root, but on any node as well. In that case it will do lookup for that branch only.

To do this, we need to add ``xpath`` also to the ``XmlNodeJsObject`` class.